explainability-core/src/main/java/org/kie/trustyai/explainability/local/counterfactual/score/DefaultCounterfactualScoreCalculator.java

```diff
@@ -198,24 +198,36 @@ private BendableBigDecimalScore calculateOutputScore(CounterfactualSolution solu
         for (PredictionOutput predictionOutput : predictions) {
 
             final List<Output> outputs = predictionOutput.getOutputs();
+            final int numberOutputs = outputs.size();
 
-            if (goal.size() != outputs.size()) {
-                throw new IllegalArgumentException("Prediction size must be equal to goal size");
-            }
+            if (solution.getGoalCriteria() == null) {
 
-            final int numberOutputs = outputs.size();
-            for (int i = 0; i < numberOutputs; i++) {
-                final Output output = outputs.get(i);
-                final Output goalOutput = goal.get(i);
-                final double d =
-                        DefaultCounterfactualScoreCalculator.outputDistance(output, goalOutput, solution.getGoalThreshold());
-                outputDistance += d * d;
-
-                if (output.getScore() < goalOutput.getScore()) {
-                    tertiaryHardScore -= 1;
+                if (goal.size() != outputs.size()) {
+                    throw new IllegalArgumentException("Prediction size must be equal to goal size");
+                }
+
+                for (int i = 0; i < numberOutputs; i++) {
+                    final Output output = outputs.get(i);
+                    final Output goalOutput = goal.get(i);
+                    final double d =
+                            DefaultCounterfactualScoreCalculator.outputDistance(output, goalOutput,
+                                    solution.getGoalThreshold());
+                    outputDistance += d * d;
+
+                    if (output.getScore() < goalOutput.getScore()) {
+                        tertiaryHardScore -= 1;
+                    }
+                }
+                primaryHardScore -= Math.sqrt(outputDistance);
```

Comment:
Is it possible to implement this with a "default" `CounterfactualGoalCriteria` implementation? Currently we have this if/else scenario but it should be nice to unify everything inside the criterial goals
```java
            if (solution.getGoalCriteria() == null) {
                ....
            } else {
                final GoalScore score = solution.getGoalCriteria().apply(outputs);
                ....
            }
```

Feel free to postpone to a different ticket if you agree with the comment but the impact is too big

Reply:
@danielezonca A `DefaultCounterfactualCriteria` is now implemented.